### ISSUE TITLE
Improve Documentation for ValidationCallback to Assist with Hyperparameter Tuning

### ISSUE THREAD

**Issue by TheRealSalmon** [deepchem/deepchem#2849]
## 🚀 Feature

I took a closer look at ValidationCallback in deepchem/models/callbacks.py and I don't think any more code needs to be written. There is already an attribute named self._best_score and an argument named save_on_minimum which allows choosing minimize or maximize. 

I have a few ideas: 1) Add a function to ValidationCallback called GetBestScore() or something. It may honestly not be necessary as like I said, self._best_score is already there. 2) Edit the dc.hyper.GridHyperparamOpt tuner to use the best validation score. I think it currently evaluates the FINAL model on the validation set instead of the BEST model but please correct me if I'm wrong. Then document this change clearly so others who wish to create a new hyperparameter optimizer will have an example to follow.

I think as a follow up, actual early stopping where the model saves the best model and stops if it doesn't improve after n steps would be a really nice feature to have, although I'm sure it'd be very complicated to implement across so many models. 

## Motivation

Allowing users to access the best validation score for the purpose of early stopping is very important for hyperparameter optimization. This feature is already there but the question is how do we make it easier to use/develop with. 

## Additional context
